File: app/src/main/java/com/halil/ozel/moviedb/ui/home/adapters/SearchHistoryAdapter.java
package com.halil.ozel.moviedb.ui.home.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.halil.ozel.moviedb.R;
import java.util.List;

public class SearchHistoryAdapter extends RecyclerView.Adapter<SearchHistoryAdapter.HistoryViewHolder> {

    private  List<String> historyList;
    private final OnHistoryItemClickListener listener;
    // Interface untuk menangani klik
    public interface OnHistoryItemClickListener {
        void onHistoryItemClick(String query);
    }

    public SearchHistoryAdapter(List<String> historyList, OnHistoryItemClickListener listener) {
        this.historyList = historyList;
        this.listener = listener;
    }
    public void updateData(List<String> newHistory) {
        this.historyList = newHistory;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public HistoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_search_history, parent, false);
        return new HistoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull HistoryViewHolder holder, int position) {
        String query = historyList.get(position);
        holder.tvQuery.setText(query);
        holder.itemView.setOnClickListener(v -> listener.onHistoryItemClick(query));
    }

    @Override
    public int getItemCount() {
        return historyList.size();
    }

    static class HistoryViewHolder extends RecyclerView.ViewHolder {
        TextView tvQuery;

        HistoryViewHolder(View itemView) {
            super(itemView);
            tvQuery = itemView.findViewById(R.id.tvHistoryQuery);
        }
    }
}

File: app/src/main/java/com/halil/ozel/moviedb/ui/search/SearchHistoryAdapter.kt
package com.halil.ozel.moviedb.ui.search

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.halil.ozel.moviedb.R

class SearchHistoryAdapter(
    private var historyList: MutableList<String>,
    private val onItemClick: (String) -> Unit,
    private val onDeleteClick: (String) -> Unit
) : RecyclerView.Adapter<SearchHistoryAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val tvHistory: TextView = view.findViewById(R.id.tv_history_text) // Pastikan ID ini ada di layout row
        val btnDelete: ImageView = view.findViewById(R.id.btn_delete_history) // Ikon silang/hapus
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        // Kita pakai layout sederhana saja atau buat row_search_history.xml
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_keyword, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val query = historyList[position]
        holder.tvHistory.text = query

        // Klik biasa untuk mencari
        holder.itemView.setOnClickListener { onItemClick(query) }

        // Klik lama untuk menghapus (Sesuai request)
        holder.itemView.setOnLongClickListener {
            onDeleteClick(query)
            true
        }

        // Atau klik ikon silang jika Anda menambahkannya di layout
        holder.btnDelete.setOnClickListener { onDeleteClick(query) }
    }

    override fun getItemCount(): Int = historyList.size

    fun updateData(newData: List<String>) {
        historyList.clear()
        historyList.addAll(newData)
        notifyDataSetChanged()
    }
}